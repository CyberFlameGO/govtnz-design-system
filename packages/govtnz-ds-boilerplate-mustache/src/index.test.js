const fs = require("fs");
const prettier = require("prettier");
const Mustache = require("mustache");

function getImportPath(importPath) {
  return (
    require.resolve
      .paths(importPath)
      .filter(nodeModule => fs.existsSync(nodeModule + "/" + importPath))[0] +
    "/" +
    importPath
  );
}

const TemplateFieldsetBlockWithError = fs.readFileSync(
  getImportPath("@govtnz/ds/build/mustache/FieldsetBlockWithError.mustache"),
  { encoding: "utf-8" }
);
const TemplateH1 = fs.readFileSync(
  getImportPath("@govtnz/ds/build/mustache/H1.mustache"),
  {
    encoding: "utf-8"
  }
);
const TemplateRadios = fs.readFileSync(
  getImportPath("@govtnz/ds/build/mustache/Radios.mustache"),
  { encoding: "utf-8" }
);
const TemplateRadioBlock = fs.readFileSync(
  getImportPath("@govtnz/ds/build/mustache/RadioBlock.mustache"),
  { encoding: "utf-8" }
);

const rawHTML = `${Mustache.render(TemplateFieldsetBlockWithError, {
  legend: `${Mustache.render(TemplateH1, {
    "styleSize=large": true,
    styleSize: true,
    id: `nameChangeId5`,
    children: `Have you changed your name?`
  })}`,
  hint: `<p >This includes changing your last name or spelling your name differently.</p>`,
  hintId: `hintId5`,
  errorId: `errorId5`,
  error: `Select 'yes' if you have changed your name.`,
  children: `${Mustache.render(TemplateRadios, {
    inline: true,
    children: `${Mustache.render(TemplateRadioBlock, {
      label: `Yes`,
      radioId: `anyRadioId9234`,
      value: `true`,
      name: `nameChange5`,
      labelId: `labelId9`,
      hintId: `hintId5`
    })}${Mustache.render(TemplateRadioBlock, {
      label: `No`,
      radioId: `anyRadioId10`,
      value: `false`,
      name: `nameChange5`,
      hintId: `hintId5`,
      labelId: `labelId10`
    })}`
  })}`
})}`;

const html = prettier.format(rawHTML, { parser: "html" });

describe("HTML Snapshot", () => {
  it("Simple global rules", async () => {
    expect(html).toMatchSnapshot();
  });

  it("Rendered templates", async () => {
    expect(TemplateFieldsetBlockWithError).toMatchSnapshot();
    expect(TemplateH1).toMatchSnapshot();
    expect(TemplateRadios).toMatchSnapshot();
    expect(TemplateRadioBlock).toMatchSnapshot();
  });
});
