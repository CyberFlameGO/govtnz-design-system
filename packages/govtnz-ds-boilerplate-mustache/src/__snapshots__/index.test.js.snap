// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Snapshot Rendered templates 1`] = `
"{{! DEVELOPER NOTE: This template uses triple-bracket \\"{{{\\"
    which disables HTML escaping.
    Please ensure these variables are properly escaped:
     - legend,
     - hint,
     - error,
     - children.
    The reason for this is to allow raw HTML, for values such as (eg) '<span lang=\\"mi\\">M훮ori</span>'. }}

<div
class=\\"g-fieldsetBlock-form-group{{#errorId}} g-fieldsetBlock-form-group--error{{/errorId}}\\"
>
  
  
<fieldset
aria-describedby=\\"{{hintId}} {{errorId}}\\"
class=\\"g-fieldset\\"
>
  
    
<legend
class=\\"g-fieldset__legend\\"
>
  
      
  {{{legend}}}
  
    
</legend>
  
    
  
      
<div
class=\\"g-fieldsetBlock-hint\\"
{{#hintId}}
  id=\\"{{hintId}}\\"
{{/hintId}}
>
  
        
  {{{hint}}}
  
      
</div>
  
    
  
    
  
      
<div
class=\\"g-fieldsetBlock-error-message\\"
{{#errorId}}
  id=\\"{{errorId}}\\"
{{/errorId}}
>
  
        
<span
class=\\"g-fieldsetBlock-visually-hidden\\"
>
  
          Error:
        
</span>
  
        
  {{{error}}}
  
      
</div>
  
    
  
    
<div
>
  {{{children}}}
</div>
  
  
</fieldset>
  

</div>"
`;

exports[`HTML Snapshot Rendered templates 2`] = `
"{{! DEVELOPER NOTE: This template uses triple-bracket \\"{{{\\"
    which disables HTML escaping.
    Please ensure these variables are properly escaped:
     - children.
    The reason for this is to allow raw HTML, for values such as (eg) '<span lang=\\"mi\\">M훮ori</span>'. }}

<h1
class=\\"{{#styleSize=xlarge}} g-heading-xl{{/styleSize=xlarge}}{{#styleSize=large}} g-heading-l{{/styleSize=large}}{{#styleSize=medium}} g-heading-m{{/styleSize=medium}}{{#styleSize=small}} g-heading-s{{/styleSize=small}}{{#styleSize=xsmall}} g-heading-xs{{/styleSize=xsmall}}{{#styleSize=xxsmall}} g-heading-xxs{{/styleSize=xxsmall}} {{#marginBottom8}} g-heading-mb-8{{/marginBottom8}} {{#marginBottom0}} g-heading-mb-0{{/marginBottom0}}\\"
{{#id}}
  id=\\"{{id}}\\"
{{/id}}
>
  
  
  {{{children}}}
  

</h1>"
`;

exports[`HTML Snapshot Rendered templates 3`] = `
"{{! DEVELOPER NOTE: This template uses triple-bracket \\"{{{\\"
    which disables HTML escaping.
    Please ensure these variables are properly escaped:
     - children.
    The reason for this is to allow raw HTML, for values such as (eg) '<span lang=\\"mi\\">M훮ori</span>'. }}

<div
class=\\"g-radios{{#inline}} g-radios--inline{{/inline}}\\"
>
  
  
  {{{children}}}
  

</div>"
`;

exports[`HTML Snapshot Rendered templates 4`] = `
"{{! DEVELOPER NOTE: This template uses triple-bracket \\"{{{\\"
    which disables HTML escaping.
    Please ensure these variables are properly escaped:
     - label,
     - hint.
    The reason for this is to allow raw HTML, for values such as (eg) '<span lang=\\"mi\\">M훮ori</span>'. }}

<div
class=\\"g-radios__item\\"
>
  
  
<input
{{#hintId}}
  aria-describedby=\\"{{hintId}}\\"
{{/hintId}}
class=\\"g-radios__input\\"
{{#radioId}}
  id=\\"{{radioId}}\\"
{{/radioId}}
{{#name}}
  name=\\"{{name}}\\"
{{/name}}
type=\\"radio\\"
{{#disabled}}
  disabled=\\"{{#disabled}}disabled{{/disabled}}\\"
{{/disabled}}
{{#readOnly}}
  readonly=\\"{{#readOnly}}readOnly{{/readOnly}}\\"
{{/readOnly}}
{{#autoFocus}}
  autofocus=\\"{{#autoFocus}}autoFocus{{/autoFocus}}\\"
{{/autoFocus}}
{{#value}}
  value=\\"{{value}}\\"
{{/value}}
{{#checked}}
  checked=\\"{{#checked}}checked{{/checked}}\\"
{{/checked}}
/>
  
  
<label
class=\\"g-radioBlock-label g-radios__label\\"
{{#radioId}}
  for=\\"{{radioId}}\\"
{{/radioId}}
>
  
    
  {{{label}}}
  
  
</label>
  
  
  
    
<div
class=\\"g-radioBlock-hint g-checkboxes__hint\\"
{{#hintId}}
  id=\\"{{hintId}}\\"
{{/hintId}}
>
  
      
  {{{hint}}}
  
    
</div>
  
  
  

</div>"
`;

exports[`HTML Snapshot Simple global rules 1`] = `
"<div class=\\"g-fieldsetBlock-form-group g-fieldsetBlock-form-group--error\\">
  <fieldset aria-describedby=\\"hintId5 errorId5\\" class=\\"g-fieldset\\">
    <legend class=\\"g-fieldset__legend\\">
      <h1 class=\\" g-heading-l  \\" id=\\"nameChangeId5\\">
        Have you changed your name?
      </h1>
    </legend>

    <div class=\\"g-fieldsetBlock-hint\\" id=\\"hintId5\\">
      <p>
        This includes changing your last name or spelling your name differently.
      </p>
    </div>

    <div class=\\"g-fieldsetBlock-error-message\\" id=\\"errorId5\\">
      <span class=\\"g-fieldsetBlock-visually-hidden\\">
        Error:
      </span>

      Select 'yes' if you have changed your name.
    </div>

    <div>
      <div class=\\"g-radios g-radios--inline\\">
        <div class=\\"g-radios__item\\">
          <input
            aria-describedby=\\"hintId5\\"
            class=\\"g-radios__input\\"
            id=\\"anyRadioId9234\\"
            name=\\"nameChange5\\"
            type=\\"radio\\"
            value=\\"true\\"
          />

          <label
            class=\\"g-radioBlock-label g-radios__label\\"
            for=\\"anyRadioId9234\\"
          >
            Yes
          </label>

          <div class=\\"g-radioBlock-hint g-checkboxes__hint\\" id=\\"hintId5\\"></div>
        </div>
        <div class=\\"g-radios__item\\">
          <input
            aria-describedby=\\"hintId5\\"
            class=\\"g-radios__input\\"
            id=\\"anyRadioId10\\"
            name=\\"nameChange5\\"
            type=\\"radio\\"
            value=\\"false\\"
          />

          <label class=\\"g-radioBlock-label g-radios__label\\" for=\\"anyRadioId10\\">
            No
          </label>

          <div class=\\"g-radioBlock-hint g-checkboxes__hint\\" id=\\"hintId5\\"></div>
        </div>
      </div>
    </div>
  </fieldset>
</div>
"
`;
